# A Novel Measure of Non-coding Genome Conservation Identifies Genomic Regulatory Blocks Within Primates
### Alexander J. Nash<sup>1, 2</sup> and Boris Lenhard<sup>1, 2, *</sup>
<sup>1</sup>Computational Regulatory Genomics Group, MRC London Institute of Medical Sciences, Du Cane Road, London W12 0NN, UK, <sup>2</sup>Institute of Clinical Sciences, Faculty of Medicine, Imperial College London, Hammersmith Campus, Du Cane Road, London W12 0NN, UK.
<sup>*</sup>To whom correspondence should be addressed. 
## Abstract
__Motivation:__ Clusters of extremely conserved non-coding elements (CNEs) mark genomic regions devoted to cis-regulation of key developmental genes in Metazoa. We have recently show that their span coincides with that of topologically associating domains (TADs), making them useful for estimating conserved TAD boundaries in the absence of Hi-C data. The standard approach - detecting CNEs in genome alignments and then establishing the boundaries of their clusters - requires tuning of several parameters and breaks down when comparing closely related genomes.
__Results:__ We present a novel, kurtosis-based measure of pairwise non-coding conservation that requires no pre-set thresholds for conservation level and length of CNEs. We show that it performs robustly across a large span of evolutionary distances, including across the closely related genomes of primates for which standard approaches fail. The method is straightforward to implement and enables detection and comparison of clusters of CNEs and estimation of underlying TADs across a vastly increased range of Metazoan genomes.




## Scripts and data used for publication

General kurtosis scripts are located in the base directory, while all Data is found in /Data/
